.github/workflows/integration.yml

```diff
@@ -40,94 +43,346 @@ jobs:
-          - test_name: 'shutdown'
-            if: ${{ github.event_name == 'merge_group' || needs.changes.outputs.int-all == 'true' }}
-          - test_name: 'splunk'
-            if: ${{ github.event_name == 'merge_group' || needs.changes.outputs.int-all == 'true' || needs.changes.outputs.splunk == 'true' }}
-          - test_name: 'webhdfs'
-            if: ${{ github.event_name == 'merge_group' || needs.changes.outputs.int-all == 'true' || needs.changes.outputs.webhdfs == 'true' }}
-
-  # This is a required status check, so it always needs to run if prior jobs failed, in order to mark the status correctly.
-  integration:
+    if: |
+      github.event_name == 'merge_group' || (
+        needs.changes.outputs.all-int == 'true'
+          || needs.changes.outputs.amqp == 'true'
+          || needs.changes.outputs.appsignal == 'true'
+          || needs.changes.outputs.aws == 'true'
+          || needs.changes.outputs.axiom == 'true'
+          || needs.changes.outputs.azure == 'true'
+          || needs.changes.outputs.clickhouse == 'true'
+          || needs.changes.outputs.databend == 'true'
+          || needs.changes.outputs.datadog == 'true'
+          || needs.changes.outputs.dnstap == 'true'
+          || needs.changes.outputs.docker-logs == 'true'
+          || needs.changes.outputs.elasticsearch == 'true'
+          || needs.changes.outputs.eventstoredb == 'true'
+          || needs.changes.outputs.fluent == 'true'
+          || needs.changes.outputs.gcp == 'true'
+          || needs.changes.outputs.humio == 'true'
+          || needs.changes.outputs.http-client == 'true'
+          || needs.changes.outputs.influxdb == 'true'
+          || needs.changes.outputs.kafka == 'true'
+          || needs.changes.outputs.logstash == 'true'
+          || needs.changes.outputs.loki == 'true'
+          || needs.changes.outputs.mongodb == 'true'
+          || needs.changes.outputs.nats == 'true'
+          || needs.changes.outputs.nginx == 'true'
+          || needs.changes.outputs.opentelemetry == 'true'
+          || needs.changes.outputs.postgres == 'true'
+          || needs.changes.outputs.prometheus == 'true'
+          || needs.changes.outputs.pulsar == 'true'
+          || needs.changes.outputs.redis == 'true'
+          || needs.changes.outputs.splunk == 'true'
+          || needs.changes.outputs.webhdfs == 'true'
+      )
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v3
+
+      - run: sudo npm -g install @datadog/datadog-ci
+
+      - run: docker image prune -af ; docker container prune -f
+
+      - name: Create test script
+        run: |
+          echo "
+              cargo vdev -v int test -a \$1
+              RET=\$?
+              cargo vdev -v int stop \$1
+              ./scripts/upload-test-results.sh
+              exit \$RET
+          " > int_test.sh
+          cat int_test.sh
```

Comment:
This test script is created twice in workflows. Are there any downsides to just checking it in to `scripts`?

Also, what is the purpose of the `cat` at the end?

Reply:
I'm not to familiar with guidelines on what can go into `scripts/` . If its OK to drop things like this that are 100% only supposed to run in CI then def, I'd be happy to move that there.

The `cat` at the end is just to allow debugging the workflow. It's present throughout the regression workflow as well. It's useful to be able to see that when debugging issues in the workflow and at least know "ok that part looks correct". It's how I noticed I was missing a couple escapes.

Reply:
Looks like there are CI specific scripts there. I'll move it.